app/services/audit/wasapi_warc_lister.rb

```diff
@@ -11,17 +11,18 @@ def initialize(wasapi_collection_id:, wasapi_provider:, wasapi_account:, embargo
       @embargo_months = embargo_months
     end
 
+    # Iterate through each WARC filename, or return as a list
     def each(&block)
-      if block_given?
-        next_url = first_page_url
-        while next_url
-          response = get_page(next_url)
-          files_for(response).each { |file| block.call(file) }
-          next_url = response[:next]
-        end
-      else
-        to_enum(:each)
-      end
+      return to_enum(:each) unless block_given?
+
+      files(yield_object: false, &block)
+    end
+
+    # Iterate through WARC file metadata, or return as a list
```

Comment:
Can you make it clearer what is returned by this method (i.e., what is WARC file metadata)?